### ISSUE TITLE
Issue with reproducibility

### ISSUE THREAD

**Issue by RushikeshJoshi4** [vitchyr/rlkit#102]
I'm trying to train a model such that it can be reproduced again. I tried setting the seed parameter in the run_experiment function, however, I am not able to get the same result again. Is there something I'm missing here?

More details:
Training SAC according to the examples/sac.py
Using mujoco environments. I am currently working on the Humanoid v3.
Everything else is quite similar the the examples/sac.py file.

**Comment by vitchyr** [comment]
I realized that [I don't set the random seed for pytorch](https://github.com/vitchyr/rlkit/blob/f136e140a57078c4f0f665051df74dffb1351f33/rlkit/launchers/launcher_util.py#L336), so perhaps that's why you're getting inconsistent result. Perhaps setting the seed there will help get consistent results.

**Comment by RushikeshJoshi4** [comment]
Sorry for the delayed reply. Yes Vitchyr, you're right. 
Just for reference of those are trying to create reproducible results:
1. Make sure set torch.manual_seed() where the seed is set for numpy and random.
2. Also, make sure you set the seeds appropriately for the environments too. Otherwise, the environment will give you random observations. But you can't do this just after you create the environment. Otherwise, every time the environment is reset (in the rollout function), it will give you same observations. Instead you want to set the environment seeds differently for different replay of each of different epochs. So, make sure that is set appropriately.

Closing this issue since I realised my mistake.
